src/screens/Gurbani/index.tsx

```diff
@@ -1,23 +1,37 @@
 import React from 'react'
+import { StackScreenProps } from '@react-navigation/stack'
 import { useQuery } from 'react-query'
 
-import Screens from '../../lib/screens'
+import Screens, { AppStackParams } from '../../lib/screens'
+import { ContentTypes, LineData } from '../../types/data'
+import { getBani, getShabad } from '../../data'
 import Container from '../../components/Container'
-import { getShabad } from '../../data/shabads'
 
 import BottomBar from './BottomBar'
 import Lines from './Lines'
 import Navbar from './Navbar'
 
-type ShabadQuery = { queryKey: [string] }
-const shabadQuery = ( { queryKey }: ShabadQuery ) => getShabad( ...queryKey )
+type Loaders = {
+  [screen in ContentTypes]: ( options : { queryKey: [string] } ) => Promise<{ lines: LineData[] }>
+}
+
+// Loaders all return a common interface. Is there a better way to deal with specifics of each type?
```

Comment:
In [Flow](https://flow.org/en/) they have [`$ObjMap`](https://flow.org/en/docs/types/utilities/#toc-objmap). In TypeScript there is not an exact same thing but something somewhat similar is achievable.

Reply:
What's the difference between that and what I have now?

Reply:
It is doing the same thing. I though you want a  template that is more generic and infer types